src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +637,7210 @@ ActorProfile Boss_07_Profile = {
+
+    if (Animation_OnFrame(&this->skelAnime, this->animEndFrame)) {
+        Boss07_Wrath_ChooseJump(this, play, true);
+    }
+}
+
+void Boss07_Wrath_SetupDamaged(Boss07* this, PlayState* play, u8 damage, u8 dmgEffect) {
+    if ((s8)this->actor.colChkInfo.health >= 0) {
+        this->actor.colChkInfo.health -= damage;
+    }
+
+    if ((s8)this->actor.colChkInfo.health <= 0) {
+        if (KREG(19) != 0) {
+            Audio_PlaySfx_AtPos(&sMajoraSfxPos, NA_SE_EN_LAST3_VOICE_DEAD_OLD);
+        } else {
+            Actor_PlaySfx(&this->actor, NA_SE_EN_LAST3_VOICE_DEAD_OLD);
+        }
+
+        this->shouldStartDeath = true;
+        Enemy_StartFinishingBlow(play, &this->actor);
+    } else {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_LAST3_VOICE_DAMAGE2_OLD);
+
+        if (this->actionFunc != Boss07_Wrath_Damaged) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &gMajorasWrathDamagedAnim, -10.0f);
+            this->actionFunc = Boss07_Wrath_Damaged;
+            this->animEndFrame = Animation_GetLastFrame(&gMajorasWrathDamagedAnim);
+            this->sfxTimer = 0;
+        } else if (dmgEffect == MAJORAS_WRATH_DMGEFF_ANIM_FRAME_CHECK) {
+            if (this->skelAnime.curFrame <= (this->animEndFrame - 5.0f)) {
+                this->disableCollisionTimer = 30;
+            } else {
+                Animation_MorphToPlayOnce(&this->skelAnime, &gMajorasWrathDamagedAnim, -5.0f);
+            }
+        }
+    }
+}
+
+void Boss07_Wrath_Damaged(Boss07* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    this->sfxTimer++;
+
+    if ((this->sfxTimer % 16) == 0) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_LAST2_WALK2_OLD);
+    }
+
+    Boss07_SmoothStop(this, 2.0f);
+
+    this->leftWhip.mobility = this->rightWhip.mobility = 0.7f;
+    this->leftWhip.gravity = this->rightWhip.gravity = -15.0f;
+    this->leftWhip.drag = this->rightWhip.drag = 2.0f;
+    this->leftWhip.tension = this->rightWhip.tension = 0.0f;
+
+    if (Animation_OnFrame(&this->skelAnime, this->animEndFrame)) {
+        Boss07_Wrath_ChooseJump(this, play, true);
+    }
+}
+
+void Boss07_Wrath_CheckWhipCollisions(Vec3f* whipPos, f32 tension, Boss07* this, PlayState* play) {
+    s32 i;
+    PlayerImpactType playerImpactType = -1;
+    Player* player = GET_PLAYER(play);
+    f32 dx;
+    f32 dy;
+    f32 dz;
+    Vec3f hitPos;
+
+    if ((tension >= 50.0f) && (this->whipCollisionTimer == 0)) {
+        Actor* prop = play->actorCtx.actorLists[ACTORCAT_PROP].first;
+
+        while (prop != NULL) {
+            if (prop->id == ACTOR_OBJ_TSUBO) {
+                for (i = 10; i < sWhipLength; i += 3) {
+                    dx = prop->world.pos.x - whipPos[i].x;
+                    dy = prop->world.pos.y + 10.0f - whipPos[i].y;
+                    dz = prop->world.pos.z - whipPos[i].z;
+
+                    if (sqrtf(SQ(dx) + SQ(dy) + SQ(dz)) < (KREG(38) + 60.0f)) {
+                        ((ObjTsubo*)prop)->unk_19B = 1;
```

Comment:
In `z_obj_tsubo.c`, it's currently not a bool. Look at line 476 in the latest `main`:
```c
    } else if ((this->unk_19B != 0) ||
               (acHit && (this->cylinderCollider.elem.acHitElem->atDmgInfo.dmgFlags & 0x058BFFBC))) {
        typeData = &sPotTypeData[type];
        this->unk_19B = 0;
```
It looks like the only values that this is set to are 0 and 1, though, so I changed it in response to your review. I just wanted to leave a paper trail of this edit in case someone feels really strongly about how this variable should be represented in the pot actor.